### PR TITLE
fix: exclude geospatial ops from pandas/dask/polars `has_operation`

### LINKED ISSUE #4699: docs: operation support matrix incorrectly states pandas/dask support geospatial operations
The [operation support matrix](https://ibis-project.org/docs/3.2.0/backends/support_matrix/) currently incorrectly states that pandas/dask support geospatial operations.

This is because:

- The `gen_matrix` [script](https://github.com/ibis-project/ibis/blob/master/gen_matrix.py) uses `backend.has_operation` to check if an operation is supported for a backend.
- The `pandas`/`dask` backends implement this method by checking if the operation type (or a parent class of that type) are registered by the `execute_node`/`pre_execute` dispatch functions.
- The current implementation of the pandas/dask backends register a single handler for base classes like `ops.Binary` rather than individual handlers for subclasses like `ops.Add`
- The geospatial operations also subclass from these base classes, and are thus mistakenly identified as supported

The "proper" fix for this is to probably rework the pandas `execute_node` dispatch registry to avoid registering handlers for base classes like `Binary`, instead registering handlers for each individual binary op.

A more pragmatic fix would be to add a hardcoded exception to either `has_operation` or the `gen_matrix` script to mark the geospatial operations as unsupported.

### PR DESCRIPTION
**Pull request by jcrist**
Fixes #4699.

### CODE REVIEW

File: ibis/backends/dask/tests/test_client.py [hunk 1]
```diff
@@ -136,3 +135,9 @@ def test_invalid_connection_parameter_types(npartitions):
     )
     with pytest.raises(TypeError, match=expeced_msg):
         ibis.dask.from_dataframe("file.csv")
+
+
+def test_has_operation(client):
```
Comment (cpcloud) [inline ibis/backends/dask/tests/test_client.py:140]: Any reason not to make this a backend test? Seems simple enough to `never` the ones that do support it.

Comment (jcrist) [reply ibis/backends/dask/tests/test_client.py:140]: Bug felt specific enough to these three backends. Happy to move the test if you think that'd be better.

Comment (cpcloud) [reply ibis/backends/dask/tests/test_client.py:140]: Let's put this into a single backend test.

**Comment by cpcloud** [comment]
Friendly ping @jcrist!

File: ibis/backends/dask/tests/test_client.py (hunk 1 above)
Comment (jcrist) [reply ibis/backends/dask/tests/test_client.py]: Done

**Review by cpcloud — APPROVED** [review]
Thanks!

**Pull request merged**